test/integration/features/basic.feature

```diff
@@ -27,6 +31,8 @@ Feature: Basic
       hello
       """
 
+  Scenario: Minishift --docker-env works properly
```

Comment:
@hferentschik I agree best placement for this test will be in docker.feature in future and that proper scenario for this begins with minishift start and then check if env is there. Until then it can be here I think even if it does not fit into feature well. Your naming 

> "A user can set custom Docker specific environment variables"

seems fine to me.